src/main.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console, @typescript-eslint/no-explicit-any */
+export const LogLevels = {
+  DEBUG: 10,
+  INFO: 20,
+  WARNING: 30,
+  ERROR: 40,
+};
+
+type Tags = Record<string, any>;
+
+export class ProperLogger {
+  name: string;
+
+  commonTags: Tags;
+
+  logLevel: number;
+
+  console: Pick<Console, 'debug' | 'log' | 'warn' | 'error'>;
+
+  static getLogger(name: string): ProperLogger {
+    return new ProperLogger(name);
+  }
+
+  constructor(name: string) {
```

Comment:
Is it intentional that we don't have constructor optional arguments for `commonTags` and `logLevel`?

Reply:
Ah good question!

`commonTags` is an omission. The Python `ProperLogger` has it as a constructor argument so this should too.

The `logLevel` is not present because it's a little hard to reason about how that would interact with the environment variable. I think that if we included it here it would be a little ambiguous which between the argument and the environment variable which would have precedence. By excluding it from the constructor we're sacrificing some amount of flexibility for a simpler interface.

Will add `commonTags` to the constructor.